### ISSUE TITLE
[BUG] Sitemap plugin fails on export

### ISSUE THREAD

**Issue by simonunlikely** [react-static/react-static#1233]
Hi,

It seems sitemap generation via `react-static-sitemap-plugin` works well on `yarn build` but fails on `yarn export` giving the following error : 

```
 Error: Trace: Error: The sitemap url prefix cannot be empty or undefined!
   
```

`siteRoot` is defined in my `static.config.js`

Any clues ?

Thanks !

**Comment by SleeplessByte** [comment]
Did you set `REACT_STATIC_PUBLIC_PATH`' environment variable?

**Comment by julesce** [comment]
I'm experiencing a related problem (I suspect). I have the `REACT_STATIC_PUBLIC_PATH` env variable set, and `yarn build` works as expected, generating the correct `sitemap.xml`.

However, when I run `yarn export`, I end up with a `sitemap.xml` where the `siteRoot` value from `static.config.js` is essentially missing, or treated as an empty string.

An example, under `yarn build`:

    <url><loc>https://www.hello.com/company/</loc></url>

And `yarn export`:

    <url><loc>/company/</loc></url>

I'll do some further digging, and hopefully yield some more information or a pull request.

**Comment by benedfit** [comment]
There are a number of environmental variables (https://github.com/newhighsco/destiny-clan-warfare/blob/master/plugins/plugin-netlify-cache/node.api.js#L84-L98) that are set in the webpack step of `build` and `bundle` that aren't set when running `export` on its own

**Comment by chejroo** [comment]
Bug still exist in app after `react-static create`

**Comment by SleeplessByte** [comment]
There is no new version released since #1352 was merged 4 days ago. Please always check CHANGELOG.md when you're trying to figure out _what's_ in your version.
